### ISSUE TITLE
FB Authentication is not working for me!

### ISSUE THREAD

**Issue by cayasso** [ciaranj/connect-auth#45]
Hi, the fallowing code doesnt work, (response from FB doesnt authenticate), when I go to localhost.com:3000 (I have it configured like this in my hosts file) and I get redirected to facebook asking for login, I enter my information and then it returns me to /auth/facebook but req.isAuthenticated() keep returning FALSE.

I do have the correct fbId and secret key. Inf fact its working with mongoose-auth but I really want to get connect-auth working for me.

Thank you in advance for any help!

<pre>
var fbId= "xxxxxxxxxxxxx";
var fbSecret= "xxxxxxxxxxxxxx";
var fbCallbackAddress= "http://localhost.com:3000/auth/facebook"

var cookieSecret = "node";     // enter a random hash for security

var express= require('express');
var auth = require('connect-auth')
var app = express.createServer();

app.configure(function(){
    app.use(express.cookieParser());
    app.use(express.session({secret: cookieSecret}));
    app.use(auth([
        auth.Facebook({
            appId : fbId, 
            appSecret: fbSecret, 
            callback: fbCallbackAddress, 
            scope: 'email'
        })
    ]));
});

// Method to handle a sign-in with a specified method type, and a url to go back to ...
app.get('/', function(req,res, params) {

  req.authenticate(['facebook'], function(error, authenticated) { 
    
    console.log('STATUS: ' + authenticated);
        
    if (authenticated ) {
      console.log('AUTHENTICATED');
    }
    else {      
     console.log('NOT AUTHENTICATED');
    }
    
   });
});

app.get('/auth/facebook', function(req,res) {
    console.log(req.isAuthenticated());
});
app.listen(3000);
</pre>

**Comment by dz0ny** [comment]
Try:

```
var fbId= "xxxxxxxxxxxxx";
var fbSecret= "xxxxxxxxxxxxxx";
var fbCallbackAddress= "http://localhost.com:3000/auth/facebook"

var cookieSecret = "node";     // enter a random hash for security

var express= require('express');
var auth = require('connect-auth')
var app = express.createServer();


app.configure(function(){
    app.use(express.bodyParser());
    app.use(express.methodOverride());
    app.use(express.cookieParser());
    app.use(express.session({secret: cookieSecret}));
    app.use(auth([
        auth.Facebook({
            appId : fbId, 
            appSecret: fbSecret, 
            callback: fbCallbackAddress, 
            scope: 'email'
        })
    ]));
    app.use(app.router);
});


app.get('/auth/facebook', function(req, res) {
  req.authenticate("facebook", function(error, authenticated) {
    if (authenticated) {
      res.redirect("/great");
    }else{
      res.redirect("/fail");
    }
  });
})  
```

**Comment by frostbytten** [comment]
Please try: frostbytten/connect-auth@eea4b62d2724f4ca58300a4360bcc3d0206b6b26 for fixes to FBAuth.

If you want to have a custom redirect (as above going to "/fail"):

`````` auth.Facebook({
  appId: fbId,
  appSecret: fbSecret,
  callback: fbCallbackAddress,
  scope: 'email',
  failedUri: '/fail'})```

and remove:
```} else {
  res.redirect("/fail");
}```
``````

**Comment by ciaranj** [comment]
Implemented in 0.3.0
